### PR TITLE
[Bug] Fix bug that routine load may lost some data

### LINKED ISSUE #5095: [Bug] Routine load lose data
**Describe the bug**
When the cluster load is high, the routine load will lose some data of kafka.

 - Version [0.13.12]

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

In the previous implementation, whether a subtask is in commit or abort state,
we will try to update the job progress, such as the consumed offset of kafka.
Under normal circumstances, the aborted transaction does not consume any data,
and all progress is 0, so even we update the progress, the progress will remain
unchanged.
However, in the case of high cluster load, the subtask may fail half of the execution on the BE side.
At this time, although the task is aborted, part of the progress is updated.
Cause the next subtask to skip these data for consumption, resulting in data loss.

Bad case:
LoadedRows = 0;
comittedOffset > 0
txn: aborted

Also modify some log level and config

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

- [] I have create an issue on (Fix #5095 ), and have described the bug/feature there in detail

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/KafkaProgress.java [hunk 1]
```diff
@@ -38,6 +42,8 @@
  */
 // {"partitionIdToOffset": {}}
 public class KafkaProgress extends RoutineLoadProgress {
+    private static final Logger LOG = LogManager.getLogger(KafkaProgress.class);
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/KafkaProgress.java:45]:
```suggestion
    private static final Logger LOG = LogManager.getLogger(RoutineLoadProgress.class);
```

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -1049,8 +1053,9 @@ private void executeTaskOnTxnStatusChanged(RoutineLoadTaskInfo routineLoadTaskIn
                                           + " maybe task was aborted by master when timeout")
                                   .build());
             }
-        } else if (checkCommitInfo(rlTaskTxnCommitAttachment, txnState.getTransactionStatus())) {
+        } else if (checkCommitInfo(rlTaskTxnCommitAttachment, txnState)) {
             // step2: update job progress
+            // only committed task need to update progress
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java:1058]: error commit...

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by hf200012 — COMMENTED** [review]
After the patch is applied, there is no data loss problem so far.

**Review by EmmyMiao87 — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**